xmlhttp.php

```diff
@@ -435,8 +435,8 @@
 	// Fetch the post from the database.
 	$post = get_post($mybb->get_input('pid', MyBB::INPUT_INT));
 
-	// No result, die.
-	if(!$post)
+	// No result or thread deleted, die.
+	if(!$post || $post['visible'] == -1)
```

Comment:
'Restore" button malfunction, to replace with this line:
if(!$post || ($post['visible'] == -1 && $mybb->input['action'] != "restorepost"))